app/models/api_action.rb

```diff
@@ -26,6 +26,7 @@ class ApiAction < ApplicationRecord
   validates :http_method, inclusion: { in: ApiAction::HTTP_METHODS}, allow_blank: true
   
   validates :payload_mapping, safe_executable: true
+  validates :custom_headers, safe_executable: true
```

Comment:
@alis-khadka , all dynamic attributes are safe by default. You don't need to validate it here. The validation for payload_mapping can also be removed.